modules/account-lib/src/coin/baseCoin/ed25519KeyPair.ts

```diff
@@ -1,22 +1,93 @@
+import * as nacl from 'tweetnacl';
+import * as hex from '@stablelib/hex';
+import { Seed, generateAccountFromSeed, generateAccount } from 'algosdk';
+import * as Crypto from '../../utils/crypto';
 import { BaseKeyPair } from './baseKeyPair';
 import { AddressFormat } from './enum';
+import { isPrivateKey, isPublicKey, isSeed, DefaultKeys, KeyPairOptions, ByteKeys } from './iface';
 import { NotImplementedError } from './errors';
 
+const DEFAULT_SEED_SIZE_BYTES = 32;
+
 export abstract class Ed25519KeyPair implements BaseKeyPair {
-  getAddress(format?: AddressFormat): string {
-    throw new NotImplementedError('getAddress not implemented');
+  protected keyPair: DefaultKeys;
+  protected source?: KeyPairOptions;
+
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param {KeyPairOptions} source Either a master seed, a private key (extended or raw), or a public key
+   */
+  protected constructor(source?: KeyPairOptions) {
+    let naclKeyPair;
+    if (!source) {
+      const seed = nacl.randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      naclKeyPair = nacl.sign.keyPair.fromSeed(seed);
+      this.setKeyPair(naclKeyPair);
+    } else if (isSeed(source)) {
+      if (Crypto.isStellarSeed(source.seed.toString())) {
+        this.keyPair = this.generateKeyPair(source.seed);
+      } else {
+        naclKeyPair = nacl.sign.keyPair.fromSeed(new Uint8Array(source.seed));
+        this.setKeyPair(naclKeyPair);
+      }
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
   }
 
-  getKeys(): any {
-    // TODO: implement from stellar seed, etc.
-    throw new NotImplementedError('getKeys not implemented');
+  private setKeyPair(naclKeyPair: nacl.SignKeyPair): void {
+    this.keyPair = {
+      prv: hex.encode(naclKeyPair.secretKey).slice(0, 64),
+      pub: hex.encode(naclKeyPair.publicKey),
+    };
+  }
+
+  abstract getAddress(format?: AddressFormat): string;
+
+  getKeys(): DefaultKeys | ByteKeys {
+    const result: DefaultKeys = {
+      pub: this.keyPair.pub,
+    };
+
+    if (this.keyPair.prv) {
+      result.prv = this.keyPair.prv;
+    }
+
+    return result;
   }
 
   recordKeysFromPrivateKey(prv: string): void {
-    throw new NotImplementedError('recordKeysFromPrivateKey not implemented');
+    if (!Crypto.isValidEd25519Key(prv)) {
```

Comment:
This is not going to work for keys in the protocol format.

Reply:
Do you mean in a bytes Array format (Uint8Array)?